components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/CarbonAppResource.java

```diff
@@ -56,18 +78,128 @@ public boolean invoke(MessageContext messageContext) {
+                        Utils.createJsonError("Unsupported HTTP method, " + httpMethod + ". Only GET , " +
+                                "POST and  " + "DELETE methods are supported",
+                        axis2MessageContext, BAD_REQUEST));
+                break;
+            }
+        }
         axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
         return true;
     }
 
+    private void handlePost(org.apache.axis2.context.MessageContext axisMsgCtx) {
+        JSONObject jsonResponse = new JSONObject();
+        String contentType = axisMsgCtx.getProperty(Constants.CONTENT_TYPE).toString();
+        if (!contentType.contains(MULTIPART_FORMDATA_DATA_TYPE)) {
+            JSONObject response = Utils.createJsonError("Supports only for the Content-Type : "
+                    + MULTIPART_FORMDATA_DATA_TYPE, axisMsgCtx, BAD_REQUEST);
+            Utils.setJsonPayLoad(axisMsgCtx, response);
+            return;
+        }
+
+        StringBuilder unMovedCApps = new StringBuilder();
+        String errorMessage = "Error when deploying the Carbon Application : ";
+        OMElement messageBody = axisMsgCtx.getEnvelope().getBody().getFirstElement();
+        Iterator iterator = messageBody.getChildElements();
+        boolean isDeployedSuccesfully = true;
+        while (iterator.hasNext()) {
+            OMElement fileElement = (OMElement) iterator.next();
+            String fileName = fileElement.getAttributeValue(new QName("filename"));
+            if (fileName != null && fileName.endsWith(".car")) {
+                byte[] bytes = Base64.getDecoder().decode(fileElement.getText());
+                Path cAppDirectoryPath = Paths.get(Utils.getCarbonHome(), "repository", "deployment",
+                        "server", "carbonapps", fileName);
+                try {
+                    Files.write(cAppDirectoryPath, bytes);
+                } catch (IOException e) {
+                    isDeployedSuccesfully = false;
+                    unMovedCApps.append(fileName).append(", ");
+                    log.error(errorMessage + fileName, e);
+                }
+                log.info("Successfully added Carbon Application : " + fileName);
+            } else {
+                isDeployedSuccesfully = false;
+                unMovedCApps.append((fileName != null ? fileName : "filename: <null>")).append(", ");
+            }
+        }
+
+        if (isDeployedSuccesfully) {
+            jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, "Successfully added Carbon Application(s)");
+            Utils.setJsonPayLoad(axisMsgCtx, jsonResponse);
+        } else {
+            Utils.setJsonPayLoad(axisMsgCtx, Utils.createJsonErrorObject(errorMessage + unMovedCApps));
+        }
+    }
+
+    private void handleDelete(MessageContext messageContext, org.apache.axis2.context.MessageContext axisMsgCtx) {
+        String cAppNamePattern = Utils.getPathParameter(messageContext, CAPP_NAME_PATTERN);
```

Comment:
If we are planning to support a pattern it should be a query parameter and I think it is not safe. It is correct to have a path param locating a single resource (in this case, a single CApp) and the result should be deleting exactly that Capp.